File: training/c/src/domain/model/no_such_creature_exception.c
#include <stdio.h>
#include <string.h>

#include "../../../include/de/sots/cellarsandcentaurs/domain/model/no_such_creature_exception.h"

NoSuchCreatureException no_such_creature_exception_new(CreatureId id)
{
    NoSuchCreatureException exception;
    exception.id = id;
    snprintf(exception.message, sizeof(exception.message), "%s%s", "No such creature in the dungeon: ", id.id);
    return exception;
}

const char *no_such_creature_exception_message(const NoSuchCreatureException *exception)
{
    return exception->message;
}

File: sample-projects/csharp/src/De.Sots.CellarsAndCentaurs/Domain/Model/NoSuchCreatureException.cs
using System;

namespace De.Sots.CellarsAndCentaurs.Domain.Model;

public class NoSuchCreatureException : Exception
{
    public CreatureId Id { get; }

    public NoSuchCreatureException(CreatureId id)
        : base("No such creature in the dungeon: " + id.Value)
    {
        Id = id;
    }
}
